src/puppet-padchat/padchat-manager.ts

```diff
@@ -221,7 +221,7 @@ export class PadchatManager extends PadchatRpc {
 
     const succeed = await this.tryAutoLogin(this.memorySlot)
     if (!succeed) {
-      this.startCheckScan()
+      await this.startCheckScan()
```

Comment:
This might block the start() for a long time which you will not want to, please test it before send to PR.

I would like to suggest you to try using `this.startCheckScan().catch(e => log.warn('...'))` and it might be able to work.